tensorflow_addons/callbacks/accuracy_stopping.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Callback that stops training when a specified amount of accuracy has achieved."""
+
+
+import tensorflow as tf
+from tensorflow.keras.callbacks import Callback
+from typeguard import typechecked
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class AccuracyStopping(Callback):
+    """Stop training when a specified accuracy is reached.

```

Comment:
Would we also want to extend this to other metrics as well? If you have a highly imbalanced class (for example 99.9% of training data is negative), then it would be easy to reach high accuracy just by predicting everything as negative. In this case then Precision / Recall / AUC would be a better metrics.

Reply:
@shun-lin  thanks for reviewing, yes in imbalance data accuracy is not good measure metric. So should I change my accuracy metric with Precision / Recall / AUC or create another callbacks for them. Also if user wants to use this callback then they must have one of Precision / Recall / AUC as their metric.

Reply:
You can see how earlyStopping in TensorFlow core implement this functionality (they allow the user to specify which metric to monitor), hope this helps!

Reply:
Thank you so much for helping me, will make changes soon 😊